feat: Add "require-param-name" rule

mrmlnc opened the pull request:
Hello, @gajus,

My team uses [Ramda](http://ramdajs.com), as a functional library. Unfortunately, we forget to write the Identifier's of the @param tag, because the code is the following:

```js
const getSections = R.propOr([], 'sections');
```

In most cases JSDoc looks like:

```js
/*
 * @param {Object}
 * @retrurns {Object[]}
 */
const getSections = R.propOr([], 'sections');
```

We want in the near future to move to TypeScript. But now we use it for code validation based on JSDoc comments and TypeScript think that we have not correct JSDoc (he's right).

Valid JSDoc:

```js
/*
 * @param {Object} document
 * @retrurns {Object[]}
 */
const getSections = R.propOr([], 'sections');
```

Well, with this rule we can rectify this situation.

gajus commented on README.md:
```diff
@@ -22,6 +22,7 @@ JSDoc linting rules for ESLint.
         * [`require-description-complete-sentence`](#eslint-plugin-jsdoc-rules-require-description-complete-sentence)
         * [`require-example`](#eslint-plugin-jsdoc-rules-require-example)
         * [`require-hyphen-before-param-description`](#eslint-plugin-jsdoc-rules-require-hyphen-before-param-description)
+        * [`require-identifier`](#eslint-plugin-jsdoc-rules-require-identifier)
```
You've updated generated README file. See https://github.com/gajus/eslint-plugin-jsdoc/tree/master/.README.

mrmlnc replied:
Oh, sorry. Will be fixed after fully review from your side 👍

gajus replied:
All the rest is good. :-)

mrmlnc replied:
Fixed and divided into two commits 🎉

anton-rudeshko commented on .README/rules/require-identifier.md:
```diff
@@ -0,0 +1,10 @@
+### `require-identifier`
+
+Requires that all function parameters has identifiers.
```
have

anton-rudeshko commented on src/rules/requireIdentifier.js:
```diff
@@ -0,0 +1,20 @@
+
+  _.forEach(jsdocParameters, (jsdocParameter) => {
+    if (jsdocParameter.tag && jsdocParameter.name === '') {
+      report('There must be a identifier after @param ' + (jsdocParameter.type === '' ? 'type' : 'tag') + '.');
```
**an** identifier

vitkarpov commented on README.md:
```diff
@@ -829,6 +840,13 @@ function quux () {
 }
 
 /**
+ * Тест.
```
What is this supposed to mean? Is it in Chinese? ;)

mrmlnc commented:
@gajus, I agree with @anton-rudeshko. So, I renamed rule from `require-identifier` to `require-param-name` for consistency between other rules.

Also I added a quote to description of this rule from JSDoc documentation:

> The `@param` tag requires you to specify the name of the parameter you are documenting. You can also include the parameter's type, enclosed in curly brackets, and a description of the parameter.
>
> [JSDoc](http://usejsdoc.org/tags-param.html#overview)

gajus commented:
Ready to be merged?

mrmlnc commented:
@gajus, you can release a new version of this plugin (can wait, but why not)? Maybe tomorrow or the day after :)
